### ISSUE TITLE
Second monitor Taskbar clock becomes misaligned over time.

### ISSUE THREAD

**Issue by Psyda** [White-Tiger/T-Clock#281]
Second Display: https://i.imgur.com/pYRxkAz.png

Primary Display: https://i.imgur.com/frKuCNv.png

---

I like my taskbars slim and out of the way, but with easy access to time/date. App Works great for that, but over time randomly as I tab in and out of fullscreen apps, Tclock becomes misaligned until the process and explorer are restarted.

The longer it goes on, the more it slides out of view until it can't be seen anymore.

Should note that it doesn't happen every time.

I have no idea what could be causing this one, but I've noticed windows explorer occasionally refreshes or reloads and my taskbar blinks for a moment. During that, it may be messing with T-Clock, 

I think it's worth noting that; Settings>Clock Text> Size and Text position> "horizontal setting".
It doesn't move based on the position of the taskbar's bottom. If I set it to 5 pixels, my monitor on the left moves it to the right 5px, while the monitor on the right also goes the same direction, always shoving one off of the screen, making my setup difficult to adjust.

This issue has been plaguing me for months. Requiring a restart of Explorer and T-Clock to fix.

I could upload my .ini file if that would help.

Edit; I'm able to repeat the glitch by resetting the graphics driver with the hotkey: Ctrl+Win+Shift+B

**Comment by charmarkk** [comment]
Having the same problem - hoping there are some current maintainers or folks interested. Hate to see this app die off.

**Comment by White-Tiger** [comment]
Did you try the nightly version already? https://github.com/White-Tiger/T-Clock#downloads-binaries 
There's been a lot of injection / position changes since beta. Nothing new per se (as all of this happened in 2018) but I assume it would have fixed your problem.
Alas, it works for me...

Positioning errors always happened since the Win 10 Anniversary update which required some hacky workarounds. So it might still not work in the end.

**Comment by charmarkk** [comment]
> Did you try the nightly version already? https://github.com/White-Tiger/T-Clock#downloads-binaries There's been a lot of injection / position changes since beta. Nothing new per se (as all of this happened in 2018) but I assume it would have fixed your problem. Alas, it works for me...
> 
> Positioning errors always happened since the Win 10 Anniversary update which required some hacky workarounds. So it might still not work in the end.

I'm on the beta, I'll give the nightly a try. (Never really knew there was a difference.) Still hoping for some new blood. 😅

@White-Tiger It's looking like the nightly might've fixed it, I haven't noticed the problem yet today.

**Comment by charmarkk** [comment]
@White-Tiger I spoke too soon - the problem is still happening with the Nightly version.

**Comment by White-Tiger** [comment]
But you don't know how to reproduce it? I assume it might happen after hours of use?
Does `right click->Refresh T-Clock` do anything when it's off set?

**Comment by charmarkk** [comment]
Yes, refreshing it fixes it. I assume it's just an "over time" thing, like
OP said. It MAY happen/begin when switching applications, I only notice it
after not looking at it for a while when playing games etc., and after a
while I don't see the clock.

But the display resolution etc. don't change when I do that, so unless it's
something to do with fullscreen apps, that might not be.

**Comment by charmarkk** [comment]
@White-Tiger I forgot to mention this from a while ago: I disabled ClearType (or rather, changed the Text Quality to Default - either way ClearType is off) and I _think_ that fixed it on my Win10 machine. I haven't noticed the need to reset the clock lately.

**Comment by White-Tiger** [comment]
I'll test that... I've personally been using "ClearType Natural" but switched to the ugly XP one now just on case :P

**Comment by charmarkk** [comment]
Oh is it "ClearType WinXP+" generally not supposed to be used? Both of my machines have had that selected since install. I'll have to do some more testing too it seems.

**Comment by jerry1970** [comment]
I went through the ClearType wizard, "optimizing" the monitors, and after that the T-Clock was removed from the taskbars on the external monitors. Rebooting did not bring them back. Any ideas how I can get them back?

**Comment by charmarkk** [comment]
> I went through the ClearType wizard, "optimizing" the monitors, and after that the T-Clock was removed from the taskbars on the external monitors. Rebooting did not bring them back. Any ideas how I can get them back?

Is the regular clock there? Or is there NOTHING where the clock should be? If the former, it's possible that T-Clock got removed from your Startup folder (that has happened to me from time to time). If the latter, perhaps try opening it manually from its program folder and reconfiguring it. I've also sometimes had luck by resizing the taskbar(s) temporarily.

**Comment by jerry1970** [comment]
There is no clock at all anymore but TClock is on the main display's taskbar so it does start when Windows starts. The other taskbars do show the open window buttons as usual, just no clock.

I'll try resizing. Thanks for the feedback.

**Comment by jerry1970** [comment]
> I've also sometimes had luck by resizing the taskbar(s) temporarily.

Not sure what I did exactly but it included adjusting scaling and then resizing the taskbars and back again and the clocks are back on all taskbars! :) Thanks for the help @charmarkk !

**Comment by charmarkk** [comment]
> > I've also sometimes had luck by resizing the taskbar(s) temporarily.
> 
> Not sure what I did exactly but it included adjusting scaling and then resizing the taskbars and back again and the clocks are back on all taskbars! :) Thanks for the help @charmarkk !

No problem @jerry1970! It's an old little program that has its quirks, but still works with a little elbow grease! Kinda like that old furnace that needs a smack with a wrench now and then. :) Glad it's working for you!
